### PR TITLE
Bret Funk 1705BE Scrabble

### PR DESCRIPTION
**Pull request by brettt**
BFunk's entry for Scrabble.

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -1,18 +1,28 @@
 class Scrabble
+  attr_reader :point_values
+  def initialize
```
Comment (s-espinosa) [inline lib/scrabble.rb]: Would've preferred that you left this as a method. Functionally they operate the same, so not a big deal, but want to make sure that you start thinking of methods as things that return things.

Not an issue that I would worry too much about if you had created this from scratch (6 one way, half dozen the other), but try to get used to using the existing code as is if there's not a good reason to change it.

File: lib/scrabble.rb [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+          "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+          "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+          "Y"=>4, "Z"=>10
+        }
+  end
 
   def score(word)
-    1
+    word_total = 0
+    if word.nil? || word.empty?
```
Comment (s-espinosa) [inline lib/scrabble.rb]: Like everything about this check. The boolean logic, the use of `#nil`, and the use of `#empty?` all make this very clean/readable.

File: lib/scrabble.rb [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
+      return 0
+    else
+      word.upcase.chars.each do |letter|
+        word_total += @point_values[letter.upcase]
```
Comment (s-espinosa) [inline lib/scrabble.rb]: Note that here you're not using the `attr_reader`, but instead accessing the instance variable directly.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
+        }
+  end
 
   def score(word)
-    1
+    word_total = 0
+    if word.nil? || word.empty?
+      return 0
+    else
+      word.upcase.chars.each do |letter|
```
Comment (s-espinosa) [inline lib/scrabble.rb]: Looks like a good implementation to me!

File: lib/scrabble.rb [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
+          "E"=>1, "F"=>4, "G"=>2, "H"=>4,
+          "I"=>1, "J"=>8, "K"=>5, "L"=>1,
+          "M"=>3, "N"=>1, "O"=>1, "P"=>3,
+          "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+          "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+          "Y"=>4, "Z"=>10
+        }
+  end
 
   def score(word)
```
Comment (s-espinosa) [inline lib/scrabble.rb]: Like the implementation here, but if you had some time to refactor would prefer that you had broken your conditional into some smaller methods. Could have a high level `#score` method that showed the high level steps, and then leave the nitty gritty for the supporting methods.

File: test/scrabble_test.rb [hunk 6]
```diff
@@ -1,13 +1,45 @@
...
 require 'pry'
 
 class ScrabbleTest < Minitest::Test
   def test_it_can_score_a_single_letter
-    skip
     assert_equal 1, Scrabble.new.score("a")
     assert_equal 4, Scrabble.new.score("f")
   end
+
+  def test_two_letter_word
```
Comment (s-espinosa) [inline test/scrabble_test.rb]: Test suite looks nice! Breaking into smaller methods would allow you to add some unit tests.

**Comment by brettt** [comment]
Thanks for the feedback.

Yeah I am an idiot. I kept trying to use the score method with the letters in brackets so the method wouldn't work and I moved it to an instance variable to make it work the way I knew how. I realized my incompetence afterwards.

Breaking it up into more methods is great feedback.  Thanks.

Did you see you got a shout out in my tests?!



_____________________________

**Comment by brettt** [comment]
One last thing, the method only worked if I accessed the instance variable directly. I initially had it run though attr_reader but it wouldn't run correctly. I had this same issue with CompleteMe. I am sure I made some rookie mistake so this is something I really want to improve upon in the coming weeks.  Your feedback is well received!

________________________________

**Comment by s-espinosa** [comment]
Ha! I did see the shoutout. Pretty entertaining.

Let's see if we can debug the instance variable sometime next week. 👍

**Pull request closed without merging**